File: mentees/Bipin7777777/Tests/Problem1.java


public class Problem1 {
    //public static HashMap<Integer,Integer> map = new HashMap<Integer,Integer>();
	
    public static void main(String[] args) {
		System.out.println("nos of ways will be :"+climb(5) );

	}

	
	public static int  climb(int n) {
	  int x=1;  // default for climbing 1 stairs at a time
	  
	   if(n%2==0)
	   {
		   x++; // for climbing 2 steps at a time
		   
	   }
	   else {
		   x= x+n;   //for n nos of places can be 1st step and rest will be 2nd step
	   }

		return x;
	}
}

File: Durgesh/data_structures/Problem1.py
# Q.  Distinct ways can you climb to the top with 1 or 2 steps at a time.

# Approach:
#           using recursion combine result of climbing 1 step and climbing 2 steps.

# Optimal Approach:
#                using DP to store the result of subproblems

t = int(input())

for _ in range(t):
    n = int(input())
    
    def rec(n):
        if n==1:
            return 1
        if n==0:
            return 1
        if n<0:
            return 0

        return rec(n-1) + rec(n-2)
    
    print(rec(n))


# test cases -> solution
# 2 -> 2
# 3 -> 3
# 4 -> 5
# 6 -> 13
